Курсовая работа 3-го курса Приложение socksAppWebStore Доработка от 21.02.2023

ruslankorotkov opened the pull request:
Приложение socksAppWebStore
Доработка от 21.02.2023
Так получше?

Commit:
Приложение socksAppWebStore
Доработка от 18.02.2023

Commit:
Приложение socksAppWebStore
Доработка от 20.02.2023

Commit:
Приложение socksAppWebStore
Доработка от 20.02.2023

Commit:
Приложение socksAppWebStore
Доработка от 20.02.2023

Commit:
Приложение socksAppWebStore
Доработка от 20.02.2023
Базовый уровень.

Commit:
Приложение socksAppWebStore
Доработка от 20.02.2023
Базовый уровень.

Commit:
Приложение socksAppWebStore
Доработка от 21.02.2023
Базовый уровень.

Commit:
Приложение socksAppWebStore
Доработка от 21.02.2023
Базовый уровень.

Commit:
Приложение socksAppWebStore
Доработка от 21.02.2023

Commit:
Приложение socksAppWebStore
Доработка от 21.02.2023

Commit:
Приложение socksAppWebStore
Доработка от 21.02.2023

Commit:
Приложение socksAppWebStore
Доработка от 21.02.2023
Подскажите пожалуйста. Как мне запустить чтение файла.

Commit:
Приложение socksAppWebStore
Доработка от 21.02.2023

Commit:
Приложение socksAppWebStore
Доработка от 21.02.2023

Commit:
Приложение socksAppWebStore
Доработка от 21.02.2023
Доработка.
Так лучше?

Commit:
Приложение socksAppWebStore
Доработка от 21.02.2023
Дима помоги пожалуйста доделать.
Файл сохраняется но прочесть не могу.
PostConstruct не работает.
И метод (public void readFromFile()) пишет ошибку.

Commit:
Приложение socksAppWebStore
Доработка от 21.02.2023
Дима помоги пожалуйста доделать.
Файл сохраняется но прочесть не могу.
PostConstruct не работает.
И метод (public void readFromFile()) пишет ошибку.

Commit:
Приложение socksAppWebStore
Доработка от 21.02.2023
Дима помоги пожалуйста доделать.
Файл сохраняется но прочесть не могу.
PostConstruct не работает.
И метод (public void readFromFile()) пишет ошибку.

Commit:
Приложение socksAppWebStore
Доработка от 22.02.2023
Дима помоги пожалуйста доделать.
Файл сохраняется но прочесть не могу.
PostConstruct не работает.
И метод (public void readFromFile()) пишет ошибку.

Commit:
Приложение socksAppWebStore
Доработка от 22.02.2023
Дима помоги пожалуйста доделать.
Файл сохраняется но прочесть не могу.
PostConstruct не работает.
И метод (public void readFromFile()) пишет ошибку.

Commit:
Приложение socksAppWebStore
Доработка от 23.02.2023
Дима помоги пожалуйста доделать.
Файл сохраняется но прочесть не могу.
PostConstruct не работает.
И метод (public void readFromFile()) пишет ошибку.

dmitry-bizin commented on src/main/java/sky/pro/socksappwebstore/services/impl/SocksServiceImpl.java:
```diff
@@ -0,0 +1,145 @@
+                    && socks.getSize().equals(size)
+                    && socks.getCottonPart() >= cottonMin
+                    && socks.getCottonPart() <= cottonMax) {
+                return element.getValue();
```
здесь нужно суммировать кол-во носков, подходящих под условие, не факт, что будет всегда только одна партия такая, надо суммировать

dmitry-bizin commented on src/main/java/sky/pro/socksappwebstore/services/impl/SocksServiceImpl.java:
```diff
@@ -0,0 +1,145 @@
+    public Path createAllSocks() throws IOException {
+        Path path = filesService.getSocksFile().toPath();
+        for (Map.Entry<Socks, Integer> element : socksMap.entrySet()) {
+            try (Writer writer = Files.newBufferedWriter(path, StandardOpenOption.APPEND)) {
```
файл нужно открывать один раз до цикла, а не в цикле

dmitry-bizin commented on src/main/java/sky/pro/socksappwebstore/services/impl/SocksServiceImpl.java:
```diff
@@ -0,0 +1,145 @@
+            saveToFile();
+        } else {
+            socksMap.put(socks, socksBatch.getQuantity());
+            saveToFile();
```
saveToFile у тебя в if и else, поэтому его можно вынести оттуда и вызвать один раз после if/else

dmitry-bizin commented on src/main/java/sky/pro/socksappwebstore/services/impl/SocksServiceImpl.java:
```diff
@@ -0,0 +1,145 @@
+                return socksBatch.getQuantity();
+            } else {
+                socksMap.remove(socks);
+                saveToFile();
```
saveToFile у тебя в if и else, поэтому его можно вынести оттуда и вызвать один раз после if/else

dmitry-bizin commented on src/main/java/sky/pro/socksappwebstore/services/impl/SocksServiceImpl.java:
```diff
@@ -0,0 +1,145 @@
+
+    public void saveToFile() {
+        try {
+            String json = new ObjectMapper().writeValueAsString(socksMap);
```
ObjectMapper лучше сделать полем и заинжектить через конструктор, чтобы не создавать каждый раз

dmitry-bizin commented on src/main/java/sky/pro/socksappwebstore/services/impl/SocksServiceImpl.java:
```diff
@@ -0,0 +1,145 @@
+    public void readFromFile() {
+        try {
+            String json = filesService.readFromFile();
+            socksMap = new ObjectMapper().readValue(json, new TypeReference<HashMap<Socks, Integer>>() {
```
ObjectMapper лучше сделать полем и заинжектить через конструктор, чтобы не создавать каждый раз

Commit:
Приложение socksAppWebStore
Доработка от 23.02.2023
Файл сохраняется но прочесть не могу.
PostConstruct не работает.
И метод (public void readFromFile()) пишет ошибку.
